vllm/model_executor/layers/quantization/modelopt.py

```diff
@@ -714,11 +752,38 @@ def apply(
             scoring_func=scoring_func,
             e_score_correction_bias=e_score_correction_bias)
 
+        if self.allow_flashinfer_cutlass:
```

Comment:
Swapping out `self.fused_experts` for subclasses of `FusedMoEMethodBase` should be done by an overload of `select_gemm_impl`.   This is so decisions about which communication method to use can be handled in one place in layer.py.